File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/java/io/github/lishangbu/avalon/dataset/entity/Berry.java
package io.github.lishangbu.avalon.dataset.entity;

import io.github.lishangbu.avalon.hibernate.Flex;
import jakarta.persistence.*;
import java.io.Serial;
import java.io.Serializable;
import lombok.Data;

/// 树果(Berry)实体类
///
/// 表示树果的业务信息，包括生长时间、大小、硬度等属性
///
/// @author lishangbu
/// @since 2025/08/20
@Data
@Entity
public class Berry implements Serializable {
  @Serial private static final long serialVersionUID = 1L;

  /// 主键
  @Id
  @Flex
  @Column(comment = "主键")
  private Long id;

  /// 内部名称
  @Column(comment = "内部名称", length = 100)
  private String internalName;

  /// 名称
  @Column(comment = "名称", length = 100)
  private String name;

  /// 生长到下一个阶段所需的时间(小时)
  @Column(comment = "生长到下一个阶段所需的时间(小时)")
  private Integer growthTime;

  /// 最大结果数
  @Column(comment = "最大结果数")
  private Integer maxHarvest;

  /// 树果大小（毫米）
  @Column(comment = "树果大小（毫米）")
  private Integer bulk;

  /// 光滑度
  @Column(comment = "光滑度")
  private Integer smoothness;

  /// 生长时使土壤干燥的速度，数值越高土壤干燥越快
  @Column(comment = "生长时使土壤干燥的速度，数值越高土壤干燥越快")
  private Integer soilDryness;

  /// 树果的坚硬度
  @ManyToOne(cascade = CascadeType.ALL)
  @JoinColumn(
      name = "firmness_id",
      referencedColumnName = "id",
      comment = "树果的坚硬度",
      foreignKey = @ForeignKey(name = "uk_berry_firmness_id"))
  private BerryFirmness berryFirmness;

  /// 搭配该树果使用“自然之恩”招式时继承的属性类型
  @ManyToOne(cascade = CascadeType.ALL)
  @JoinColumn(
      name = "natural_gift_type_id",
      referencedColumnName = "id",
      comment = "搭配该树果使用“自然之恩”招式时继承的属性类型",
      foreignKey = @ForeignKey(name = "uk_berry_natural_gift_type_id"))
  private Type naturalGiftType;

  /// 搭配该树果使用“自然之恩”招式时的威力
  @Column(comment = "搭配该树果使用“自然之恩”招式时的威力")
  private Integer naturalGiftPower;
}

File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/kotlin/io/github/lishangbu/avalon/dataset/entity/Berry.kt
package io.github.lishangbu.avalon.dataset.entity

import io.github.lishangbu.avalon.jimmer.id.SnowflakeIdGenerator
import org.babyfish.jimmer.jackson.JsonConverter
import org.babyfish.jimmer.jackson.LongToStringConverter
import org.babyfish.jimmer.sql.*

@Entity
@Table(name = "berry")
interface Berry {
    /** ID */
    @Id
    @GeneratedValue(generatorType = SnowflakeIdGenerator::class)
    @JsonConverter(LongToStringConverter::class)
    val id: Long

    /** 内部名称 */
    val internalName: String?

    /** 名称 */
    val name: String?

    /** 成长时间 */
    val growthTime: Int?

    /** 最大收获量 */
    val maxHarvest: Int?

    /** 体积 */
    val bulk: Int?

    /** 顺滑度 */
    val smoothness: Int?

    /** 土壤干燥度 */
    val soilDryness: Int?

    /** 树果硬度 */
    @ManyToOne
    @JoinColumn(name = "berry_firmness_id")
    val berryFirmness: BerryFirmness?

    /** 树果硬度 ID */
    @IdView("berryFirmness")
    @JsonConverter(LongToStringConverter::class)
    val berryFirmnessId: Long?

    /** 自然礼物属性 */
    @ManyToOne
    @JoinColumn(name = "natural_gift_type_id")
    val naturalGiftType: Type?

    /** 自然礼物属性 ID */
    @IdView("naturalGiftType")
    @JsonConverter(LongToStringConverter::class)
    val naturalGiftTypeId: Long?

    /** 自然礼物威力 */
    val naturalGiftPower: Int?
}
